### PR TITLE
Support TLS monerod servers and fix warning in functional-swap test

### LINKED ISSUE #396: Support tls monerod servers

### LINKED ISSUE #543: Fix warning in functional-swap test
The CheckpointEntry should be properly consumed and re-used to restore the checkpoint.

```
warning: unused import: `CheckpointEntry`
 --> tests/functional-swap.rs:8:25
  |
8 |     BitcoinFundingInfo, CheckpointEntry, MoneroFundingInfo, NodeInfo,
  |                         ^^^^^^^^^^^^^^^
  |
```

### PR DESCRIPTION
**Pull request by LeoNero**
Closes #543
Closes #396

### CODE REVIEW

File: doc/lnp/CHANGELOG.md [hunk 1]
```diff
@@ -1,56 +0,0 @@
-Change Log
```
Comment (TheCharlatan) [inline doc/lnp/CHANGELOG.md:1]: Can you do this in a separate PR?

File: Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,6 @@ include = [
   "README.md",
   "CHANGELOG.md",
   "LICENSE",
-  "config_spec.toml",
```
Comment (TheCharlatan) [inline Cargo.toml:15]: Please open a separate PR for this one as well.

File: doc/lnp/CHANGELOG.md (hunk 1 above)
Comment (LeoNero) [reply doc/lnp/CHANGELOG.md:1]: I will

File: Cargo.toml (hunk 2 above)
Comment (LeoNero) [reply Cargo.toml:15]: I will too. I am just pushing things in this PR at the moment. Will split into separate PRs later

**Comment by LeoNero** [comment]
I don't know why it marked as closed lol
